File: src/main/scala/com/bl/bd/util/ShellUtil.scala
package com.bl.bd.util

import java.io.{BufferedReader, InputStreamReader, LineNumberInputStream, LineNumberReader}

/**
  * Created by MK33 on 2016/12/7.
  */
object ShellUtil {

  def runShell(shell: String): Int = {
    val process = Runtime.getRuntime.exec(shell)
    val in = process.getInputStream
    val buffer = new BufferedReader(new InputStreamReader(in))
    val errorBuffer = new BufferedReader(new InputStreamReader(process.getErrorStream))
    var tmp = buffer.readLine()
    var tmp2 = errorBuffer.readLine()

    while (tmp != null || tmp2 != null) {
      if (tmp != null) {
        println(tmp)
        tmp = buffer.readLine()
      }
      if (tmp2 != null) {
        println(tmp2)
        tmp2 = errorBuffer.readLine()
      }
    }

    process.waitFor()
  }

}

File: src/main/java/com/xxx/rpc/test/ShellUtil.java
package com.xxx.rpc.test;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

/**
 * Created by MK33 on 2016/12/7.
 */
public class ShellUtil {

    public static void main(String args[]) throws IOException, InterruptedException {

        String s = "/opt/spark/apache-mahout-distribution-0.12.2/bin/mahout  recommenditembased  --input /tmp/kdj/shop_goods_rating  --output /tmp/kdj/mahout_recommend " +
                "--numRecommendations 2000 " +
                " --maxPrefsPerUser 1000 " +
                " --maxSimilaritiesPerItem 1000 " +
                " --similarityClassname SIMILARITY_LOGLIKELIHOOD " +
                " --tempDir /tmp/kdj/mahout_tmp ";

        Process process = Runtime.getRuntime().exec(s);
        InputStream in = process.getInputStream();
        BufferedReader buffer = new BufferedReader(new InputStreamReader(in));
        String tmp = buffer.readLine();
        while (tmp != null) {
            System.out.println(tmp);
            tmp = buffer.readLine();
        }
        process.waitFor();

    }
}
